### ISSUE TITLE
syncoid destroy my snapshot on target

### ISSUE THREAD

**Issue by piotr-sikora-v** [jimsalterjrs/sanoid#656]
Hi, I have configured my syncoid to send snapshot to second machine with FreeNAS
There I have configured periodic snapshot to keep 7 days of snapshot, but after syncoid send latest snapshot, then it delete all my snapshots od target machine.
How can I prevent this ?

**Comment by jimsalterjrs** [comment]
Syncoid must necessarily destroy any snapshots created by the target host on the target datasets. It has to do that because it must build from a common snapshot between the two.

~~~~
root@box1:~# zfs send pool/dataset@1 | ssh box2 zfs receive pool/dataset
root@box1:~# zfs snapshot pool/dataset@2
~~~~

Now you've got @1 and @2 on box1, but only have @1 on box2. If you take new snapshots directly on box2 itself...

~~~~
root@box2:~# zfs snapshot pool/dataset@my2
~~~~

You now have @1 and @2 on box1, but have @1 and @my2 on box2. If you replicate from box1 to box2:

~~~~
root@box1:~# zfs send -i pool/dataset@1 pool/dataset@2 | ssh box2 zfs receive pool/dataset
cannot receive incremental stream: pool/dataset has been modified
~~~~

In order to successfully replicate, you either need to use the -F flag on receive (which automatically destroys anything which gets in the way), or manually roll back the target to the most recent common snapshot:

~~~~
root@box1:~# ssh box2 zfs rollback -r pool/dataset@1
root@box1:~# zfs send -i pool/dataset@1 pool/dataset@2 | ssh box2 zfs receive pool/dataset
root@box1:~# ssh box2 zfs list -rt snap pool/dataset
NAME             USED  AVAIL     REFER  MOUNTPOINT
pool/dataset@1     0B      -      192K  -
pool/dataset@2     0B      -      192K  -
~~~~

... and Bob's your uncle.

In short: **you can't take snapshots locally on a backup target.** What you can do is run sanoid on the target to prune all but the most recently taken snapshots, according to policy, using `template_backup` or `template_hotspare` so that it does prune old snapshots, but does not attempt to take new ones (which, instead, replicate in from source where they were created).
